sunpy/map/map_factory.py

```diff
@@ -196,34 +197,29 @@ def _parse_args(self, *args, **kwargs):
                     data_header_pairs.append(pair)
                     i += 1    # an extra increment to account for the data-header pairing
 
-            # File name
-            elif (isinstance(arg, str) and
-                  os.path.isfile(os.path.expanduser(arg))):
-                path = os.path.expanduser(arg)
-                pairs = self._read_file(path, **kwargs)
-                data_header_pairs += pairs
-
-            # Directory
-            elif (isinstance(arg, str) and
-                  os.path.isdir(os.path.expanduser(arg))):
-                path = os.path.expanduser(arg)
-                files = [os.path.join(path, elem) for elem in os.listdir(path)]
-                for afile in files:
-                    data_header_pairs += self._read_file(afile, **kwargs)
+            # File system path (file or directory)
+            elif _is_path(arg):
+                path = pathlib.Path(arg).expanduser()
+                if path.is_file():
+                    pairs = self._read_file(path, **kwargs)
+                    data_header_pairs += pairs
+                elif path.is_dir():
+                    for afile in sorted(path.glob('*')):
+                        data_header_pairs += self._read_file(afile, **kwargs)
+                else:
+                    raise ValueError(f'{path} is neither a file nor a directory')
 
             # Glob
-            elif (isinstance(arg, str) and '*' in arg):
-                files = glob.glob(os.path.expanduser(arg))
-                for afile in files:
+            elif isinstance(arg, str) and len(glob.glob(arg)) >= 1:
```

Comment:
We can just do this right:
```suggestion
            elif isinstance(arg, str) and len(glob.glob(arg)):
```
Will `glob.glob` happily take any valid string without erroring?